toolchain/lowering/lowering_handle.cpp

```diff
@@ -24,10 +25,31 @@ auto LoweringHandleAddressOf(LoweringFunctionContext& context,
   context.SetLocal(node_id, context.GetLocal(node.GetAsAddressOf()));
 }
 
-auto LoweringHandleArrayIndex(LoweringFunctionContext& /*context*/,
-                              SemanticsNodeId /*node_id*/, SemanticsNode node)
+auto LoweringHandleArrayIndex(LoweringFunctionContext& context,
+                              SemanticsNodeId node_id, SemanticsNode node)
     -> void {
-  CARBON_FATAL() << "TODO: Add support: " << node;
+  auto [array_node_id, index_node_id] = node.GetAsArrayIndex();
+  auto* array_value = context.GetLocal(array_node_id);
+  auto* llvm_type =
+      context.GetType(context.semantics_ir().GetNode(array_node_id).type_id());
+  auto index_node = context.semantics_ir().GetNode(index_node_id);
+  llvm::Value* array_element_value;
+  if (index_node.kind() != SemanticsNodeKind::IntegerLiteral) {
+    auto* load_inst = context.builder().CreateLoad(
+        llvm_type->getArrayElementType(), context.GetLocal(index_node_id));
+    array_element_value = context.builder().CreateInBoundsGEP(
+        llvm_type, array_value,
+        context.builder().CreateZExt(
```

Comment:
Why does this need CreateZExt? Can you comment this code?

Also, please store the zext as a local, reducing function call nesting.